libvast/vast/concept/parseable/vast/time.hpp

```diff
@@ -121,6 +119,25 @@ auto const timespan = duration<vast::timespan::rep, vast::timespan::period>;
 struct ymdhms_parser : vast::parser<ymdhms_parser> {
   using attribute = timestamp;
 
+  inline sys_days to_days(unsigned short year, unsigned char month,
+                          unsigned char day) const {
+    static_assert(std::numeric_limits<unsigned>::digits >= 18,
+                  "This algorithm has not been ported to a 16 bit unsigned "
+                  "integer");
+    static_assert(std::numeric_limits<int>::digits >= 20,
+                  "This algorithm has not been ported to a 16 bit signed "
+                  "integer");
+    auto const y = static_cast<int>(year) - (month <= 2);
+    auto const m = static_cast<unsigned>(month);
+    auto const d = static_cast<unsigned>(day);
+    auto const era = (y >= 0 ? y : y - 399) / 400;
+    auto const yoe = static_cast<unsigned>(y - era * 400); // [0, 399]
+    auto const doy = (153 * (m > 2 ? m - 3 : m + 9) + 2) / 5 + d
+                     - 1;                                   // [0, 365]
+    auto const doe = yoe * 365 + yoe / 4 - yoe / 100 + doy; // [0, 146096]
+    return sys_days{} + days{era * 146097 + static_cast<int>(doe) - 719468};
```

Comment:
I'm missing a bit of documentation of where to find more information about this complex computation.